contracts/AgentMech.sol

```diff
@@ -128,25 +123,19 @@ contract AgentMech is ERC721Mech {
     mapping(address => uint256) public mapNonces;
 
     /// @dev AgentMech constructor.
-    /// @param _token Address of the token contract.
-    /// @param _tokenId The token ID.
-    /// @param _price The minimum required price.
     /// @param _mechMarketplace Mech marketplace address.
-    constructor(address _token, uint256 _tokenId, uint256 _price, address _mechMarketplace)
-        ERC721Mech(_token, _tokenId)
+    /// @param _serviceRegistry Address of the token contract.
+    /// @param _serviceId Service Id.
+    /// @param _price The minimum required price.
+    constructor(address _mechMarketplace, address _serviceRegistry, uint256 _serviceId, uint256 _price)
```

Comment:
Allow for isOperator to update its own price

Reply:
It's actually already there